cmd/darknode/darknode.go

```diff
@@ -88,35 +89,34 @@ func main() {
 
 	midpointPriceStorer := leveldb.NewMidpointPriceStorer()
 
-	// Get own nonce from leveldb, if present and store multiaddress.
-	multi, err := store.SwarmMultiAddressStore().MultiAddress(multiAddr.Address())
-	if err != nil {
-		if err != swarm.ErrMultiAddressNotFound {
-			logger.Network(logger.LevelError, fmt.Sprintf("error retrieving own nonce details from store: %v", err))
-		} else {
-			multi.Nonce = 0
-		}
-	}
-
-	multiAddr.Nonce = multi.Nonce + 1
-
 	// New crypter for signing and verification
 	crypter := registry.NewCrypter(config.Keystore, &contractBinder, 256, time.Minute)
-	multiAddrSignature, err := crypter.Sign(multiAddr.Hash())
+
+	// Insert self multiAddress if not in the storer.
+	_, err = store.SwarmMultiAddressStore().MultiAddress(multiAddr.Address())
 	if err != nil {
-		log.Fatalf("cannot sign own multiaddress: %v", err)
-	}
-	multiAddr.Signature = multiAddrSignature
-	if err := store.SwarmMultiAddressStore().InsertMultiAddress(multiAddr); err != nil {
-		log.Fatalf("cannot store own multiaddress in leveldb: %v", err)
+		if err == swarm.ErrMultiAddressNotFound {
+			signature, err := crypter.Sign(multiAddr.Hash())
+			if err != nil {
+				log.Fatalf("cannot sign own multiAddress: %v", err)
+			}
+			multiAddr.Signature = signature
+			if err := store.SwarmMultiAddressStore().InsertMultiAddress(multiAddr); err != nil {
+				log.Fatalf("cannot store own multiAddress in leveldb: %v", err)
+			}
+		} else {
+			logger.Network(logger.LevelError, fmt.Sprintf("error retrieving own multiAddress from store: %v", err))
+		}
```

Comment:
Should the dark node update its multi-address (even if it is already present in the store) to what it thinks is the most updated form? (i.e to reflect any changes in its IP address.)

Reply:
Good catch. The only way this could happen is the IP of the darknodes has been changed. It's worth checking if the IP been changed. Have pushed code to fix this